libswirl/gpl/lxdream/tacore.cpp

```diff
@@ -0,0 +1,1258 @@
+    case STATE_ERROR:
+        /* Fatal error raised - stop processing until reset */
+        return;
+
+    case STATE_EXPECT_POLY_BLOCK2:
+        /* This is always a pair of floating-point colours */
+        ta_status.intensity1 = 
+            parse_float_colour( data[0].f, data[1].f, data[2].f, data[3].f );
+        ta_status.intensity2 =
+            parse_float_colour( data[4].f, data[5].f, data[6].f, data[7].f );
+        ta_status.state = STATE_IN_LIST;
+        break;
+
+    case STATE_EXPECT_VERTEX_BLOCK2:
+        ta_parse_vertex_block2( data );
+        if( ta_status.vertex_count == ta_status.max_vertex ) {
+            ta_split_polygon();
+        }
+        break;
+
+    case STATE_EXPECT_END_VERTEX_BLOCK2:
+        ta_parse_vertex_block2( data );
+        if( ta_status.vertex_count < 3 ) {
+            ta_bad_input_error();
+        } else {
+            ta_commit_polygon();
+        }
+        ta_status.vertex_count = 0;
+        ta_status.poly_parity = 0;
+        ta_status.state = STATE_IN_LIST;
+        break;
+    case STATE_IN_LIST:
+    case STATE_IN_POLYGON:
+    case STATE_IDLE:
+        switch( TA_CMD( data->i ) ) {
+        case TA_CMD_END_LIST:
+            if( ta_status.state == STATE_IN_POLYGON ) {
+                ta_bad_input_error();
+                ta_end_list();
+                ta_status.state = STATE_ERROR; /* Abort further processing */
+            } else {
+                ta_end_list();
+            }
+            break;
+        case TA_CMD_CLIP:
+            if( ta_status.state == STATE_IN_POLYGON ) {
+                ta_bad_input_error();
+                ta_status.accept_vertexes = FALSE;
+                /* Enter stuffed up mode */
+            }
+            ta_status.clip.x1 = data[4].i & 0x3F;
+            ta_status.clip.y1 = data[5].i & 0x0F;
+            ta_status.clip.x2 = data[6].i & 0x3F;
+            ta_status.clip.y2 = data[7].i & 0x0F;
+            if( ta_status.clip.x2 >= ta_status.width )
+                ta_status.clip.x2 = ta_status.width - 1;
+            if( ta_status.clip.y2 >= ta_status.height )
+                ta_status.clip.y2 = ta_status.height - 1;
+            break;
+        case TA_CMD_POLYGON_CONTEXT:
+            if( ta_status.state == STATE_IDLE ) {
+                ta_init_list( TA_POLYCMD_LISTTYPE( data->i ) );
+            }
+
+            if( ta_status.vertex_count != 0 ) {
+                /* Error, and not a very well handled one either */
+                ta_bad_input_error();
+                ta_status.accept_vertexes = FALSE;
+            } else {
+                if( TA_IS_MODIFIER_LIST( ta_status.current_list_type ) ) {
+                    ta_parse_modifier_context(data);
+                } else {
+                    ta_parse_polygon_context(data);
+                }
+            }
+            break;
+        case TA_CMD_SPRITE_CONTEXT:
+            if( ta_status.state == STATE_IDLE ) {
+                ta_init_list( TA_POLYCMD_LISTTYPE( data->i ) );
```

Comment:
same as above in the vertex context

Reply:
Will add warning. HLE already has one.

Reply:
Looking at the logic a bit closer, this happens under normal operation - it's a check for the per-list list-init, not the global one. Global list init is handled in `lxd_ta_init`